components/match2/commons/match_group_input_util.lua

```diff
@@ -963,6 +964,34 @@ function MatchGroupInputUtil.parseParticipants(playerIds, inputPlayers, indexToP
 	return participants, unattachedParticipants
 end
 
+---@param playerIds MGIParsedPlayer[]
+---@param inputPlayers table[]
+---@param indexToPlayer fun(playerIndex: integer): PlayerInputData?
+---@param transform fun(playerIndex: integer, playerIdData: MGIParsedPlayer, playerInputData: PlayerInputData): table?
+---@return table
+function MatchGroupInputUtil.parseMapPlayers(playerIds, inputPlayers, indexToPlayer, transform)
+	local transformedPlayers = {}
+	local playerIdToIndex = Table.map(inputPlayers, function(playerIndex)
+		local playerInputData = indexToPlayer(playerIndex) or {}
+		if playerInputData.name and not playerInputData.link then
+			playerInputData.link = MatchGroupInputUtil.makeLinkFromName(playerInputData.name)
+		end
+		local playerId = MatchGroupInputUtil.findPlayerId(playerIds, playerInputData.name, playerInputData.link)
+		transformedPlayers[playerIndex] = transform(playerIndex, playerIds[playerId] or {}, playerInputData)
+		if playerId then
+			return playerId, playerIndex
+		end
+		return 0, nil
+	end)
+
+	local mappedPlayers = Array.map(playerIds, function(_, playerId)
+		local transformedPlayer = Table.extract(transformedPlayers, playerIdToIndex[playerId])
```

Comment:
```suggestion
		local transformedPlayer = Table.extract(transformedPlayers, playerIdToIndex[playerId] or '')
```
to avoid
![image](https://github.com/user-attachments/assets/dd28bed8-a797-45a9-8de7-2358a6de745c)